### PR TITLE
Dev2049/pgvector size fix

### PR DESCRIPTION
**Pull request by dev2049**
thanks @woodworker for the fix! slight update to make it possible to explicitly pass in vector size

### CODE REVIEW

File: docs/ecosystem/pgvector.md [hunk 1]
```diff
@@ -24,6 +24,10 @@ To import this vectorstore:
 from langchain.vectorstores.pgvector import PGVector
 ```
 
+PGVector embedding size is not autodetected. If you are using ChatGPT or any other embedding with 1536 dimensions
+default is fine. If you are going to use for example HuggingFaceEmbeddings you need to set the environment variable named `PGVECTOR_VECTOR_SIZE`
```
Comment (dev2049) [inline docs/ecosystem/pgvector.md:28]: ah should update

**Comment by dev2049** [comment]
looking a bit closer im now confused about what this class is used for. seems like it's usually not instantiated but just the class attributes are accessed, in which case my update to pass things into `__init__` is wrong and we should have the default value on the class access environ var. @hwchase17 does that sound about right?

**Comment by SeriyVol4ishe** [comment]
> looking a bit closer im now confused about what this class is used for. seems like it's usually not instantiated but just the class attributes are accessed, in which case my update to pass things into `__init__` is wrong and we should have the default value on the class access environ var. @hwchase17 does that sound about right?
[this class is a sqlalchemy declarative model. its used by sqlalchemy ](https://docs.sqlalchemy.org/en/20/orm/mapping_styles.html#orm-declarative-mapping)

**Comment by tarasglek** [comment]
The simpler fix and usage would be to remove vector dimension. Like in https://github.com/hwchase17/langchain/compare/master...tarasglek:langchain:taras/vector-dimensions 

I started working on my own pull req, noticed this one.

**Comment by cjcjameson** [comment]
@dev2049 I am following roughly these instructions and ran into this https://python.langchain.com/en/latest/modules/indexes/vectorstores/examples/pgvector.html

I pulled @tarasglek 's branch and it worked; this branch failed with:

```
python3.11/site-packages/langchain/vectorstores/pgvector.py", line 184, in create_tables_if_not_exists
    self._conn.commit()
    ^^^^^^^^^^^^^^^^^
AttributeError: 'Connection' object has no attribute 'commit'

```

**Pull request closed without merging**